charts/reaper-operator/templates/deployment.yaml

```diff
@@ -27,4 +30,6 @@ spec:
         name: reaper-operator
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         resources: {{- toYaml .Values.resources | nindent 10 }}
+        securityContext:
+          {{- ternary ($defaultSecurityCtx) (.Values.securityContext | toYaml) (empty .Values.securityContext) | nindent 10 }}
```

Comment:
Could this be simplified to something like this:

```
          securityContext:
            {{- toYaml .Values.securityContext | nindent 12 }}
```

Reply:
I can maybe see a simplified approach for the `pod securityContext` as it has fewer impact points throughout the `deployment` and `cassdc` for value defaults.  But there was a reason I was making the pod and container templates the same.

The reason for the template ternary empty checks on `securityContext` is that the default would need to reside in all locations of the values.yaml.   If that were to contain more properties (which we are still figuring out), this would become messy in the values file IMO.  

That complexity would apply to both pod and container securityContext.  Today, we think the default is {} for the pod securityContext, but if this were to expand to additional properties, this becomes a nice thing to handle in the helpers in one place.  Also, it would maintain consistency with the container securityContext defaults (having as {}  in the values yaml and the helper translates that to a default w/ 0..N properties).

_Examples:_

// Today's container default
```yaml
securityContext: 
  readOnlyRootFilesystem: true
  ...
```

// As an example of pod defaults other than {}
```yaml
securityContext: 
  runAsUser: 1000
  runAsGroup: 3000
  fsGroup: 2000
  ...
```

One observation that you mentioned is that we can eliminate the value defaults in the operator's `deployment` and I would need to define this in a `_helpers.tpl` in the `k8ssandra-commons` as `deployment` doesn't have access to `_helpers.tpl` in k8ssandra templates.  

That would lead me to suggest we also migrate the existing k8ssandra (used in cassdc.yaml) defaults for securityContext to the commons template helper.  We would then have one place to adjust securityContext defaults (which may grow in length) for containers and pods.  wdyt?

Reply:
👍  to adding the templates in `_helpers.tpl`.

One other thing to keep in mind. The properties available in a SecurityContext and PodSecurityContext are slightly different, so you will probably want to use separate templates them.

Reply:
💯 That is correct, we will have a default common template for pod and container security contexts.

I've pushed the common template updates, but looks to be breaking some integration tests, so once that is resolved, I'll have you verify this once more.  Thanks!